temp/legacy-code/src/main/java/com/ivy/legacy/utils/AmountFormatting.kt

```diff
@@ -195,6 +200,24 @@ fun formatInputAmount(
     return null
 }
 
+suspend fun Double.toDecimalFormat(context: Context): String {
+    val appContext = context.applicationContext ?: error("Application Context Not Found")
+    val entryPoint =
+        EntryPoints.get(appContext, FormatMoneyUseCaseImpl.FormatMoneyUseCaseEntryPoint::class.java)
+    val useCase = entryPoint.FormatMoneyUseCase()
+    return useCase.format(this)
+}
+
+fun Double.toDecimalFormatWithFraction(fraction: Int = 2): String {
```

Comment:
Why do we need this? Let's use only the FormatMoneyUseCase. My comment is about fixing its decimal format

Reply:
@ILIYANGERMANOV  Removed this change. Now we see `".0"` in place of  `".00"` again. Please note that this issue is only limited to preview only. In actual app it works as expected.
I tried to use `FormatMoneyUseCase` for preview. So, I created a non-suspending version of  ` format() ` [here](https://github.com/shamim-emon/ivy-wallet/blob/24bdfee01906480f50a2077f7214072d4204678f/shared/ui/core/src/main/java/com/ivy/ui/FormatMoneyUseCase.kt#L38-L44) but in such case we can't get preview where it's used(refer to sc below)
![Capture](https://github.com/user-attachments/assets/297e8cab-ba3c-46bd-b05d-3b279e1b44e4)

So for preview, going back to normal double to string conversion.
Please let me know your comment on this.
Thanks

Reply:
How would you recommend using the formatting(for preview) in concerned composable?

1.Passing the ViewModel function or the formatted value itself through the composable hierarchy to concernd composable?

2.Inject the viewmodel within an extension function( haven’t done it before) and call the method there and do the formatting?

Reply:
@ILIYANGERMANOV How about now? For preview we only need to make sure to show 2 decimal places right? So, how about using existing extension function `Double.format(digits)` ?
Please let me know if you see any issue with current implemented approach.

Reply:
@shamim-emon the existing function should be deleted at some point. Proper fix:
1. Provide the format usecase as composition local
2. Provide a format usecsee instance for preview

Goals:
- use only the format usecase
- don't couple the code with legacy code that we need to cleanup 

If that doesn't work, the proper but harder solution is to rework the VMs to return String instead of Double for the amount that's already formatted. That must be done at some point for performance reasons